internal_ws/yksg/src/lib.rs

```diff
@@ -44,65 +44,53 @@ impl Drop for LocalMem {
 }
 
 impl LocalMem {
-    /// Given a pointer `src` and a size, write its value to the pointer `dst`.
-    pub fn write_val(&mut self, dst: *mut u8, src: *const u8, size: usize) {
-        unsafe {
-            std::ptr::copy(src, dst, size);
-        }
-    }
-
     /// Write a constant to the pointer `dst`.
-    pub fn write_const(&mut self, dest: *mut u8, constant: &Constant) {
+    unsafe fn write_const(&mut self, dst: *mut u8, constant: &Constant) {
         match constant {
             Constant::Int(ci) => match ci {
                 ConstantInt::UnsignedInt(ui) => match ui {
-                    UnsignedInt::U8(v) => self.write_val(dest, [*v].as_ptr(), 1),
-                    UnsignedInt::Usize(v) => {
-                        let bytes = v.to_ne_bytes();
-                        self.write_val(dest, bytes.as_ptr(), bytes.len())
-                    }
+                    UnsignedInt::U8(v) => std::ptr::copy(v, dst, 1),
+                    UnsignedInt::Usize(v) => std::ptr::copy(v, dst as *mut usize, 1),
                     _ => todo!(),
                 },
                 ConstantInt::SignedInt(_) => todo!(),
             },
-            Constant::Bool(b) => self.write_val(dest, [*b as u8].as_ptr(), 1),
-            Constant::Tuple(t) => {
-                if SIR.ty(t).size() == 0 {
-                    // ZST: do nothing.
-                } else {
-                    todo!()
-                }
-            }
+            Constant::Bool(b) => {
+                // XXX are we sure that we're always supposed to copy 1 byte, and not instead to
+                // set 1 bit?
+                std::ptr::copy(b as *const bool, dst as *mut bool, 1)
```

Comment:
This just stores a bool in a variable, e.g. `$1 = true`. So, I'm fairly certain we are not supposed to set only a bit here.

Reply:
I *think* you're right, but I'm not 100% sure. Where should we check to be 100% sure?

Reply:
Setting bits, e.g. `a &= 0b0001` will result in a `BinOp:BitAnd` operation, which we haven't implemented yet. So far the interpreter has only implemented `BinOp::Add` and `BinOp::Lt`.

`write_const` is only used in `store` (and `copy_args`) which just assigns values to a local and doesn't do any binary operations.

Reply:
Does that mean setting a byte is definitely OK?

Reply:
Since `write_const` is called from `store`, rust will have ensured that the constant and the destination have the same type. So assuming `bool`'s in Rust are 1 byte, copying 1 byte is fine here.

I hope this answers your question, since I wasn't quite sure what exactly you were asking.

Reply:
I checked, and I agree https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=edb602f2e4628e4d3a6a553ad174aee2

Reply:
We chould change `write_const` to take a size as an argument, which is passed by `store`, so that we don't have to hardcode sizes in. Although, `bool` might be the only case where this is necessary, or can you think of any others?

Reply:
I don't think we need to worry further about this.